02_munge/src/munge_noaa_nos.py

```diff
@@ -30,6 +30,29 @@ def read_data(raw_datafile, read_location, s3_bucket):
         df = pd.read_csv(obj.get("Body"))
     return df
 
+def fill_gaps(x):
+    '''fills any data gaps in the middle of the input series
+    using linear interpolation; returns gap-filled time series
+    '''
+    #find nans
+    bd = np.isnan(x)
+
+    #early exit if there are no nans  
+    if not bd.any():
+        return x
+
+    #find nonnans index numbers
+    gd = np.flatnonzero(~bd)
+
+    #ignore leading and trailing nans
+    bd[:gd.min()]=False
+    bd[(gd.max()+1):]=False
```

Comment:
This is just the first and last entry correct?

Reply:
yes, still hourly! it is the same as whatever time step goes in